CMakeLists.txt

```diff
@@ -1,227 +1,175 @@
-  INPUT_FILE ${${kit}_CMAKE_SOURCE_DIR}/qSlicerAppVersionConfigure.h.in
-  OUTPUT_FILE ${CMAKE_CURRENT_BINARY_DIR}/qSlicerAppVersionConfigure.h
-  NUMBER_OF_CONFIGURE 2
+option(Slicer_BUILD_QT_DESIGNER_PLUGINS         "Build Qt designer plugins"                           OFF)
+option(Slicer_BUILD_DICOM_SUPPORT               "Build application with DICOM support"                ON)
+option(Slicer_BUILD_DIFFUSION_SUPPORT           "Build application with Diffusion support"            OFF)
+option(Slicer_BUILD_EXTENSIONMANAGER_SUPPORT    "Build application with ExtensionManager support"     OFF)
+option(Slicer_BUILD_MULTIVOLUME_SUPPORT         "Build application with MultiVolume support"          OFF)
+option(Slicer_BUILD_PARAMETERSERIALIZER_SUPPORT "Build application with parameter serializer support" OFF)
+option(Slicer_USE_PYTHONQT                      "Build application with Python support"               ON)
+option(Slicer_USE_PYTHONQT_WITH_TCL             "Build application with Python Tcl adapter layer"     OFF)
+option(Slicer_USE_QtTesting                     "Build application with QtTesting support"            OFF)
+option(Slicer_USE_SimpleITK                     "Build application with SimpleITK support"            OFF)
+
+option(Slicer_BUILD_BRAINSTOOLS                 "Build application with BRAINSTools module"           OFF)
+option(Slicer_BUILD_DataStore                   "Build application with DataStore module"             OFF)
+option(Slicer_BUILD_CompareVolumes              "Build application with ChangeTrackerPy module"       OFF)
+option(Slicer_BUILD_LandmarkRegistration        "Build application with LandmarkRegistration module"  OFF)
+
+# Enable Slicer built-in modules
+set(Slicer_CLIMODULES_ENABLED
+  LabelMapSmoothing
+  ModelMaker
+  ResampleScalarVectorDWIVolume # Needed by 'CropVolume' module
   )
-set_property(GLOBAL APPEND PROPERTY ${kit}_INCLUDE_DIRS
-  ${CMAKE_CURRENT_BINARY_DIR}
+set(Slicer_QTLOADABLEMODULES_ENABLED
+  )
+set(Slicer_QTSCRIPTEDMODULES_ENABLED
   )
 
-# --------------------------------------------------------------------------
-# Sub projects
-# --------------------------------------------------------------------------
-include(ExportLibrary)
+# Disable Slicer built-in modules
+set(Slicer_CLIMODULES_DISABLED
+  )
+set(Slicer_QTLOADABLEMODULES_DISABLED
+    #Cameras
+    #Units
+    #Terminologies
+    #Colors
+    #SubjectHierarchy
+    #Annotations
+    #Markups
+    #Transforms
+    #Data
+    #DoubleArrays
+    #Models
+    #SceneViews # Needed to avoid warning when loading scene
+    #Segmentations
+    SlicerWelcome
+    #Tables
+    #Reformat
+    #ViewControllers
+    #Volumes
+    #VolumeRendering
+  )
+set(Slicer_QTSCRIPTEDMODULES_DISABLED
+  #DataProbe
+  DMRIInstall
+  Editor
+  EditorLib
+  Endoscopy
+  LabelStatistics
+  PerformanceTests
+  SampleData
+  #ScreenCapture
+  #SegmentEditor
+  #SegmentStatistics
+  SelfTests
+  SurfaceToolbox
+  VectorToScalarVolume
+  )
```

Comment:
Would it make more sense to set the 4 commented modules in `Slicer_QTSCRIPTEDMODULES_ENABLED` instead?

Reply:
I would have to diff the current list in Slicer with the one here first. I think this is good enough for now